internal/pkg/agent/application/dispatcher/dispatcher.go

```diff
@@ -320,8 +320,10 @@ func reportNextScheduledUpgrade(input []fleetapi.Action, detailsSetter details.O
 	}
 
 	upgradeDetails := details.NewDetails(nextUpgrade.Version, details.StateScheduled, nextUpgrade.ID())
-	startTime, _ := nextUpgrade.StartTime()
-	upgradeDetails.Metadata.ScheduledAt = startTime
+	startTime, err := nextUpgrade.StartTime()
```

Comment:
we should not have an error here as we're guarding against this in a loop. 
in case we end up with an error here following lines are questionable
cc @ycombinator